common/unsafe/src/main/java/org/apache/spark/sql/catalyst/util/CollationFactory.java

```diff
@@ -176,11 +176,11 @@ public Collation(
    */
 
   public static StringSearch getStringSearch(
-      final UTF8String left,
-      final UTF8String right,
+      final UTF8String targetString,
+      final UTF8String patternString,
```

Comment:
Agreed that this should be changed, but I find it a bit weird. We call it `targetString` and then call `toString` method on it later. I would maybe add something like `targetUTF8String`. @dbatomic what do you think about this?